src/rgw/rgw_bucket.cc

```diff
@@ -1381,11 +1381,18 @@ static int bucket_stats(RGWRados *store, const std::string& tenant_name, std::st
   }
 
   utime_t ut(mtime);
+  string index_type;
+ 
+  if (bucket_info.index_type == RGWBIType_Normal) 
+    index_type = "Normal";
+  else if (bucket_info.index_type == RGWBIType_Indexless) 
+    index_type = "Indexless";
```

Comment:
It would be better to define `ostream operator<<(ostream& cout, const index_type& type)`, and just call `formatter->dump_stream("index_type") << bucket_info.index_type; ` here;